Source/WebCore/platform/network/curl/CurlProxySettings.cpp

```diff
@@ -41,6 +41,14 @@ static constexpr uint16_t SocksProxyPort = 1080;
 
 static std::optional<String> createProxyUrl(const URL&);
 
+std::optional<CurlProxySettings> CurlProxySettings::tryMakeProxySettings(Mode mode, String&& url, String&& ignoreHosts)
```

Comment:
This never returns std::nullopt it seems? Why return a std::optional?

Reply:
Oh I thought the serialization framework assumed thered be a `std::optional`. If that's not the case then I can do that.

Reply:
Not 100% sure when using `CreateUsing` in the .in file. However, you can definitely stop using `CreateUsing` and use a constructor instead. And probably should if there is no failure case.